### PR TITLE
Fix for unsafe CSE of Structs

### PR DESCRIPTION
**Pull request by briansull**
- Assign a unique value number when reading a struct field as we don't know the size of the read
- Modify the Jit Dump for CSE
- No Asm diffs in the Frameworks
- Added new test Runtime_38920.cs

### CODE REVIEW

**Comment by briansull** [comment]
Fixes #38920

File: src/tests/JIT/Regression/JitBlue/Runtime_38920/Runtime_38920.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        s.c = ++c;  // 3
+        UnsafeCastFromAPrimitiveStructFieldToStruct(s, b, c);
+
+        b = b * b;
+        c = c * c;
+        s.b = b;    // 4
+        s.c = c;    // 9
+        UnsafeCastFromAPrimitiveStructFieldToStruct(s, b, c);
+
+	if (exitStatus == 100)
```
Comment (sandreenko) [inline src/tests/JIT/Regression/JitBlue/Runtime_38920/Runtime_38920.cs:110]: Could you please format the test? VS can do it with "ctrl+K, ctrl+F".

**Review by sandreenko — APPROVED** [review]
LGTM

File: src/tests/JIT/Regression/JitBlue/Runtime_38920/Runtime_38920.cs (hunk 1 above)
Comment (briansull) [reply src/tests/JIT/Regression/JitBlue/Runtime_38920/Runtime_38920.cs:110]: I prefer my formatting over the formatting that VS would perform

Comment (briansull) [reply src/tests/JIT/Regression/JitBlue/Runtime_38920/Runtime_38920.cs:110]: I will indent the if stmt properly

**Pull request merged**